### ISSUE TITLE
Add Flower Baseline Implementation: pFedHN

### ISSUE THREAD

**Issue by achiverram28** [adap/flower#2192]
### Paper

Personalized Federated Learning using Hypernetworks

### Link

https://proceedings.mlr.press/v139/shamsian21a.html

### Maybe give motivations about why the paper should be implemented as a baseline.

We can solve Data Heterogenity issues in FL and we can build personalised models for each clients.
As per our literature survey HyperNetworks is a promising area in this research direction.

### Is there something else you want to add?

Team Information:
We are a team consisting of 2 members Ram Samarth BB(@achiverram28) and Kishan Gurumurthy(@kishan-droid) respectively from India who are federated learning and deep learning enthusiasts . We have extensive enthusiasm in going through,researching and implementing  deep learning models using TensorFlow and are eager to explore novel approaches like Personalized Federated Learning with Hypernetworks.

**Our Perspectives:**

Personalized Federated Learning with Hypernetworks: We believe that this hybrid approach holds great promise for solving the challenges of decentralized data analysis while preserving data privacy. The combination of federated learning and hypernetworks allows us to create personalized models for individual datasets, making the overall system more adaptive and efficient.

Text Data Analysis on UCI ML Repository: Analysing text datasets poses unique challenges.

We would like to reproduce the results with  MNIST, CIFAR-10, and CIFAR-100 datasets using Tensorflow and flower .

We recently contributed to flower in the form of a pull request which got merged successfully.

**How We Will Use This:**

Our team aims to implement the PFH model using TensorFlow and evaluate it on the specified text datasets from the UCI ML Repository.

### For first time contributors

- []  Read the [`first contribution` doc](https://flower.dev/docs/first-time-contributors.html)
- []  Complete the Flower tutorial
- []  Read the Flower Baselines docs to get an overview:
    - []  [https://flower.dev/docs/using-baselines.html](https://flower.dev/docs/using-baselines.html)
    - []  [https://flower.dev/docs/contributing-baselines.html](https://flower.dev/docs/contributing-baselines.html)

### Prepare - understand the scope

- [ ] Read the paper linked above
- [ ] Create the directory structure in Flower Baselines (just the `__init__.py` files and a `README.md`)
- [ ] Before starting to write code, write down all of the specs of this experiment in a README (dataset, partitioning, model, number of clients, all hyperparameters, …)
- [ ] Open a draft PR

### Implement - make it work

- [ ] Implement some form of dataset loading and partitioning in a separate `dataset.py` (doesn’t have to match the paper exactly)
- [ ] Implement the model in PyTorch
- [ ] Write a test that shows that the model has the number of parameters mentioned in the paper
- [ ] Implement the federated learning setup outlined in the paper, maybe starting with fewer clients
- [ ] Plot accuracy and loss
- [ ] Run it and check if the model starts to converge

### Align - make it converge

- [ ] Implement the exact data partitioning outlined in the paper
- [ ] Use the exact hyperparameters outlined in the paper
- [ ] Make it converge to roughly the same accuracy that the paper states
- [ ] Commit the final hyperparameters and plots
- [ ] Mark the PR as ready

**Comment by jafermarq** [comment]
@achiverram28 , i'll close this issue since we already discussed how to start with your baseline contribution. Just as a reminder, you need to follow the steps in the `Contributing a new baseline` section here: https://github.com/adap/flower/tree/main/baselines#contributing-a-new-baseline
